refactor: replace txe oracle in favor of pxe oracle

nventuro opened the pull request:
This fully deletes the `TXE` class (from `txe_oracle`). This means that `env.utility_context` and `env.private_context` are now running _the exact same code_ as real simulations using PXE. TXE's job is now much smaller: it just coordinates other components, but it doesn't have a lot of behavior of its own.

Because in Noir tests we don't allow for nested private calls, I made the `simulator` parameter in the oracle optional and simply throw an error if we attempt to call it (which we don't in txe because the translator rejects the call).

I went through the TXE oracles one by one, spotting how they matched or differred from the PXE ones, and the only relevant difference I found is that in TXE we used to immediately throw if the notify nullifier was called with a settled nullifier - which PXE does not do. Both approaches would make sense, and we want to have tests to make sure this behavior is stable, but the important thing to note is that from now on _both TXE and PXE_ will behave the same in this regard. We should try to maintain this consistency as much as possible to avoid TXE from growing into an unmaintainable useless monster. At any rate, this specific thing is handled in #17317.

Once this PR goes in, and the TXE tests indirectly become (to some extent) PXE tests as well, I'll begin opening PRs fixing the different issues I've been finding.

Thunkar reviewed (approved):
This is just brilliant

benesjan commented on yarn-project/txe/src/txe_session.ts:
```diff
@@ -11,53 +11,76 @@ import {
 
 import type { IAvmExecutionOracle, ITxeExecutionOracle } from './oracle/interfaces.js';
-import { TXE } from './oracle/txe_oracle.js';
+// import { TXE } from './oracle/txe_oracle.js';
```
```suggestion
```

Forgotten?

benesjan commented on yarn-project/txe/src/txe_session.ts:
```diff
@@ -11,53 +11,76 @@ import {
+import { makeTxEffect } from './utils/tx_effect_creation.js';
 
 /**
  * A TXE Session can be ine one of four states, which change as the test progresses and different oracles are called.
```
```suggestion
 * A TXE Session can be in one of four states, which change as the test progresses and different oracles are called.
```

benesjan reviewed (approved):
Mesmerizing

nventuro committed:
refactor: replace txe oracle in favor of pxe oracle

This fully deletes the `TXE` class (from `txe_oracle`). This means that `env.utility_context` and `env.private_context` are now running _the exact same code_ as real simulations using PXE. TXE's job is now much smaller: it just coordinates other components, but it doesn't have a lot of behavior of its own.

Because in Noir tests we don't allow for nested private calls, I made the `simulator` parameter in the oracle optional and simply throw an error if we attempt to call it (which we don't in txe because the translator rejects the call).

I went through the TXE oracles one by one, spotting how they matched or differred from the PXE ones, and the only relevant difference I found is that in TXE we used to immediately throw if the notify nullifier was called with a settled nullifier - which PXE does not do. Both approaches would make sense, and we want to have tests to make sure this behavior is stable, but the important thing to note is that from now on _both TXE and PXE_ will behave the same in this regard. We should try to maintain this consistency as much as possible to avoid TXE from growing into an unmaintainable useless monster. At any rate, this specific thing is handled in #17317.

Once this PR goes in, and the TXE tests indirectly become (to some extent) PXE tests as well, I'll begin opening PRs fixing the different issues I've been finding.
